pkg/volume/flexvolume/flexvolume.go

```diff
@@ -241,76 +245,142 @@ func (f flexVolumeMounter) GetAttributes() volume.Attributes {
 // flexVolumeManager is the abstract interface to flex volume ops.
 type flexVolumeManager interface {
 	// Attaches the disk to the kubelet's host machine.
-	attach(mounter *flexVolumeMounter) (string, error)
+	attach(mounter *flexVolumeMounter, volOptions map[string]string) (string, error)
 	// Detaches the disk from the kubelet's host machine.
 	detach(unmounter *flexVolumeUnmounter, dir string) error
 	// Mounts the disk on the Kubelet's host machine.
-	mount(mounter *flexVolumeMounter, mnt, dir string) error
+	mount(mounter *flexVolumeMounter, mnt, dir string, volOptions map[string]string) error
 	// Unmounts the disk from the Kubelet's host machine.
-	unmount(unounter *flexVolumeUnmounter, dir string) error
+	unmount(unmounter *flexVolumeUnmounter, dir string) error
 }
 
 // SetUpAt creates new directory.
 func (f *flexVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
-
-	notmnt, err := f.blockDeviceMounter.IsLikelyNotMountPoint(dir)
-	if err != nil && !os.IsNotExist(err) {
-		glog.Errorf("Cannot validate mountpoint: %s", dir)
+	err := f.Attach(f.spec, "")
```

Comment:
It becomes clearer in https://github.com/kubernetes/kubernetes/pull/26801 (see `operation_executor.go`). But the idea is that `SetupAt` is the the mount function all plugins implement. It's up to them how they implement it. Attachers get an additional mount function `MountDevice`. Once PR #26801 goes in, it ensures `MountDevice` is called to mount the device to a global path on the node. Then `Setup` is called for each pod to bind mount the pod path to the global mount path. Implementers can choose not to follow this model, and have the MountDevice be a no-op and put all their logic in `Setup`, that's up to them.